multisig-wallet/project/multisig-contract/src/data_structures.sw

```diff
@@ -1,24 +1,50 @@
 library data_structures;
 
-pub struct User {
-    // Contracts cannot sign therefore restrict scope to Address
-    /// The wallet address of a user
-    identity: Address,
-    /// Number of approvals the user provides when approving.
-    /// The default is usually 1
-    weight: u64,
+use std::{b512::B512, contract_id::ContractId, identity::Identity};
+
+pub enum MessageFormat {
+    None: (),
+    EIP191PersonalSign: (),
+}
+
+pub enum MessagePrefix {
+    None: (),
+    Ethereum: (),
+}
+
+pub enum WalletType {
+    Fuel: (),
+    EVM: (),
+}
+
+pub struct SignatureData {
+    signature: B512,
+    /// The type of formmatting of the message that was signed.
+    format: MessageFormat,
+    /// The type of prefix prepended to the message that was signed.
+    prefix: MessagePrefix,
+    /// The wallet type of the signer of the message.
+    wallet_type: WalletType,
 }
 
 pub struct Transaction {
-    /// Unique identifier for the contract which prevents this Tx from being submitted to another
-    /// instance of the multisig
+    /// Unique identifier for the contract which prevents this transaction from being submitted to another
+    /// instance of the multisig.
     contract_identifier: ContractId,
-    /// Payload sent to destination  // TODO: change to vec when implemented
+    /// Payload sent to destination  // TODO: change to Bytes when implemented: https://github.com/FuelLabs/sway/pull/3454
     data: b256,
-    /// The recipient (output / contract) regarding the Tx details
+    /// The recipient (output / contract) regarding the Tx details.
     destination: Identity,
-    /// Value used to prevent double spending
+    /// Value used to prevent double spending.
     nonce: u64,
-    /// Amount of asset
+    /// Amount of asset.
     value: u64,
 }
+
+pub struct User {
+    /// The wallet address of a user.
+    address: b256,
```

Comment:
This is something that I've noticed but it slipped my mind. Looking at the code it's been changed to save on some conversions so it should make it slightly cheaper to use and possibly deploy. The `Address` is correct thought because it prevents someone from accidentally using the `b256` of a contract as a user and then finding out that they cannot sign etc. For clarity and correctness, even at a slight cost, `Address` should be used.